src/main/java/nl/tudelft/jpacman/board/Square.java

```diff
@@ -104,7 +105,7 @@ void remove(Unit occupant) {
 	 * @return <code>true</code> iff all occupants of this square have this
 	 *         square listed as the square they are currently occupying.
 	 */
-	protected final boolean invariant() {
+	protected final boolean invariant(@UnknownInitialization(Square.class) Square this) {
```

Comment:
Here I originally tried `UnderInitialization`, which did not work.

The reason is of course that the `invariant` method is also called from methods other than the constructor, such as `link`. I did not think of the solution to replace `Under` by `Unknown`, but it makes sense of course.